### PR TITLE
Source SFTP: update descriptions in connector spec file

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What
Updated the descriptions in the SFTP source spec file to be more user-friendly.

## How
- Changed title "User Name" to "Username" for consistency
- Updated some stub descriptions to be slightly more descriptive
- Added links to the docs in optional fields and Private Key descriptions.
- Fixed a couple typos

Full list of changes can be found on the [connected issue](https://github.com/airbytehq/airbyte-internal-issues/issues/1960)!

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`85f4fd564b`](https://github.com/airbytehq/airbyte/pull/28417/commits/85f4fd564b3165fb32231886f5a2ad1865b005a6)) - ❌

⏲️  Total pipeline duration: 02mn13s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689797125/85f4fd564b3165fb32231886f5a2ad1865b005a6/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`85f4fd564b`](https://github.com/airbytehq/airbyte/pull/28417/commits/85f4fd564b3165fb32231886f5a2ad1865b005a6)) - ❌

⏲️  Total pipeline duration: 01mn30s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689797369/85f4fd564b3165fb32231886f5a2ad1865b005a6/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

File: airbyte-integrations/connectors/source-sftp/src/main/resources/spec.json [hunk 1]
```diff
@@ -8,21 +8,21 @@
     "additionalProperties": true,
     "properties": {
       "user": {
-        "title": "User Name",
-        "description": "The server user",
+        "title": "Username",
+        "description": "The username associated with the remote server.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-sftp/src/main/resources/spec.json:12]:
```suggestion
        "description": "The username to use when logging in to the remote server.",
```

**Review by sherifnada — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`3cc8342c77`](https://github.com/airbytehq/airbyte/pull/28417/commits/3cc8342c77ff4d79e576562d6f0c987bc9c45cd0)) - ❌

⏲️  Total pipeline duration: 01mn30s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689798218/3cc8342c77ff4d79e576562d6f0c987bc9c45cd0/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`3cc8342c77`](https://github.com/airbytehq/airbyte/pull/28417/commits/3cc8342c77ff4d79e576562d6f0c987bc9c45cd0)) - ❌

⏲️  Total pipeline duration: 01mn22s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689798407/3cc8342c77ff4d79e576562d6f0c987bc9c45cd0/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`60c0f91622`](https://github.com/airbytehq/airbyte/pull/28417/commits/60c0f916222ae33780431398cc51345dfb44980d)) - ❌

⏲️  Total pipeline duration: 01mn30s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689798742/60c0f916222ae33780431398cc51345dfb44980d/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`82db71be37`](https://github.com/airbytehq/airbyte/pull/28417/commits/82db71be37e72a59adf7d9f8e3c383dbeae780f1)) - ❌

⏲️  Total pipeline duration: 01mn22s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1689952886/82db71be37e72a59adf7d9f8e3c383dbeae780f1/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`7e8486bf90`](https://github.com/airbytehq/airbyte/pull/28417/commits/7e8486bf90078d6f7449f8a65fa194a5fcf01064)) - ❌

⏲️  Total pipeline duration: 02mn03s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1690226058/7e8486bf90078d6f7449f8a65fa194a5fcf01064/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`75e8be9512`](https://github.com/airbytehq/airbyte/pull/28417/commits/75e8be95126941f84213d4ddad76e080c2875c05)) - ❌

⏲️  Total pipeline duration: 02mn03s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1690297269/75e8be95126941f84213d4ddad76e080c2875c05/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`91fe1256ef`](https://github.com/airbytehq/airbyte/pull/28417/commits/91fe1256ef3721cb561731779c7e2d26ab8a557b)) - ❌

⏲️  Total pipeline duration: 02mn02s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1690297509/91fe1256ef3721cb561731779c7e2d26ab8a557b/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`1e8c109aaf`](https://github.com/airbytehq/airbyte/pull/28417/commits/1e8c109aaf9f2eeb0f62118d71c7a04afe10d415)) - ❌

⏲️  Total pipeline duration: 03mn11s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml | ✅        |
| Connector version semver check                                     | ✅        |
| Connector version increment check                                  | ✅        |
| QA checks                                                          | ✅        |
| Build connector tar                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1690298945/1e8c109aaf9f2eeb0f62118d71c7a04afe10d415/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Comment by octavia-squidington-iii** [comment]
##  source-sftp test report (commit [`f92d02b9a2`](https://github.com/airbytehq/airbyte/pull/28417/commits/f92d02b9a29307102ab992cba637f0400face816)) - ❌

⏲️  Total pipeline duration: 06mn42s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-sftp/metadata.yaml     | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |
| Build connector tar                                                    | ✅        |
| Build source-sftp docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-sftp:integrationTest | ❌        |
| Acceptance tests                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sftp-fields-update/1690300941/f92d02b9a29307102ab992cba637f0400face816/source-sftp/0.1.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sftp test
```

**Pull request closed without merging**